common/google-cloud/core/src/main/java/org/curioswitch/curiostack/gcloud/core/GcloudModule.java

```diff
@@ -61,10 +65,25 @@ public static GcloudConfig config(Config config) {
   @Provides
   @Singleton
   @GoogleApis
-  public static HttpClient googleApisClient(Optional<MeterRegistry> meterRegistry) {
+  public static HttpClient googleApisClient(
+      Optional<MeterRegistry> meterRegistry, GcloudConfig config) {
     ClientFactory factory =
         meterRegistry
-            .map(registry -> new ClientFactoryBuilder().meterRegistry(registry).build())
+            .map(
+                registry -> {
+                  ClientFactoryBuilder builder = new ClientFactoryBuilder().meterRegistry(registry);
+                  if (config.getDisableEdns()) {
+                    builder.addressResolverGroupFactory(
+                        eventLoopGroup ->
+                            new DnsAddressResolverGroup(
```

Comment:
Thanks for this change! Since making the change in server framework, armeria has been updated with a much easier API for customizing the resolver. Can you use it instead?

https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/client/ClientFactoryBuilder.java#L212